torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -422,16 +533,8 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams& params) {
     return;
   }
 
-  TensorView* reference_tv = nullptr;
-  for (auto out : output_tvs) {
-    if (out->definition() == nullptr) {
-      continue;
-    }
-    if (nRootDims(out) == max_dims) {
-      reference_tv = out;
-      break;
-    }
-  }
+  DomainMap domain_map(fusion);
+  TensorView* reference_tv = domain_map.findReferenceTensorView();
```

Comment:
This check seems wrong to me generally. The reference TV in theory should be an output that is dependent on all *resolved* broadcasts involved. That's what the check should be since there can't be any reductions, shifts, or gathers. We should also change segmentation to be based on this rule.

Reply:
What are `resolved` broadcasts?

Reply:
A broadcast resolution occurs in an expression when an input ID is a broadcast and the output ID is not a broadcast.